File: backend/app/api/agents.py
from fastapi import APIRouter

from backend.app.agents.registry import list_agents


router = APIRouter(prefix="/api/agents", tags=["agents"])


@router.get("")
async def agents() -> dict:
    available = [
        {
            "id": agent.id,
            "name": agent.name,
            "description": agent.description,
        }
        for agent in list_agents()
    ]
    return {
        "count": len(available),
        "agents": available,
        "default_mode": "auto",
    }

File: frontend/src/api/agents.ts
export type AgentSummary = {
  id: string;
  name: string;
  description: string;
};

export type AgentList = {
  count: number;
  agents: AgentSummary[];
  default_mode: "auto";
};

export async function getAgents(): Promise<AgentList> {
  const response = await fetch("/api/agents");
  if (!response.ok) {
    throw new Error(`Agent registry returned ${response.status}`);
  }
  return response.json();
}
